ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -135,197 +98,271 @@ export default class ConfirmDecryptMessage extends Component {
+          <div className="request-decrypt-message__message">
+            <div
+              className="request-decrypt-message__message-text"
+              ref={ref}
+              onScroll={onScroll}
+            >
+              {!hasDecrypted && !hasError ? txData.msgParams.data : rawMessage}
+              {hasError ? errorMessage : ''}
             </div>
-          </div>
-        </div>
-        {hasDecrypted ? (
-          <div
-            className={classnames({
-              'request-decrypt-message__message-copy': true,
-              'request-decrypt-message__message-copy--pressed':
-                copyToClipboardPressed,
-            })}
-            onClick={() => this.copyMessage()}
-            onMouseDown={() => this.setState({ copyToClipboardPressed: true })}
-            onMouseUp={() => this.setState({ copyToClipboardPressed: false })}
-          >
-            <Tooltip
-              position="bottom"
-              title={hasCopied ? t('copiedExclamation') : t('copyToClipboard')}
-              wrapperClassName="request-decrypt-message__message-copy-tooltip"
-              style={{ display: 'flex', alignItems: 'center' }}
+            <div
+              className={classnames('request-decrypt-message__message-cover', {
+                'request-decrypt-message__message-lock--pressed':
+                  hasDecrypted || hasError,
+              })}
+            />
+            <div
+              className={classnames('request-decrypt-message__message-lock', {
+                'request-decrypt-message__message-lock--pressed':
+                  hasDecrypted || hasError,
+              })}
+              onClick={(event) => {
+                decryptMessageInline(txData, event).then((result) => {
+                  if (result.error) {
+                    setHasError(true);
+                    setErrorMessage(t('decryptInlineError', [result.error]));
+                  } else {
+                    setHasDecrypted(true);
+                    setRawMessage(result.rawSig);
+                  }
+                });
+              }}
             >
-              <div className="request-decrypt-message__message-copy-text">
-                {t('decryptCopy')}
+              <div className="request-decrypt-message__message-lock__container">
+                <i className="fa fa-lock fa-lg request-decrypt-message__message-lock__container__icon" />
+                <div className="request-decrypt-message__message-lock__container__text">
+                  {t('decryptMetamask')}
+                </div>
               </div>
-              <Icon
-                name={hasCopied ? IconName.CopySuccess : IconName.Copy}
-                color={IconColor.primaryDefault}
-              />
-            </Tooltip>
+            </div>
           </div>
-        ) : (
-          <div />
-        )}
+          {hasDecrypted && !hasError && isScrollable && !isScrolledToBottom && (
+            <ButtonIcon
```

Comment:
Could this and all the scrolling state go in a local `ScrollToBottom` button?

Reply:
Done